### PR TITLE
Registry Disk user-guide update

### PR DESCRIPTION
**Pull request by davidvossel**
Registry Disk user-guide and updates.

Requires https://github.com/kubevirt/kubevirt/pull/460 to be merged first.

### CODE REVIEW

File: registry-disk.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+Registry Disks are ephemeral storage devices that can be assigned to any number
+of active Virtual Machines. This makes them an ideal tool for users who want
+to replicate a large number of Virtual Machine workloads that do not require
+persistent data. Registry Disks are commonly used in conjunction with Virtual
+Machine Replica Sets.
+
+## When Not to use a Registry Disk
+
+Registry Disks are not a good solution for any workload that requires persistent
+disks across Virtual Machine restarts, or workloads that require Virtual
```
Comment (rmohr) [inline registry-disk.md]: Regarding to restarts, if it is just a restart on the same node, will it keep the data?

File: registry-disk.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+Machine live migration support. It is possible Registry Disks may gain live
+migration support in the future, but at the moment live migrations are
+incompatible with Registry Disks. 
+
+## Registry Disk Workflow Example
+
+Users push Virtual Machine disks into the container registry using a KubeVirt
+base designed to work with the Registry Disk feature. The latest base container
+image is  kubevirt.io/registry-disk-v1alpha.
```
Comment (rmohr) [inline registry-disk.md]: could you highlight `kubevirt.io/registry-disk-v1alpha1`?

File: registry-disk.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
+## Registry Disk
+
+The Registry Disk feature provides the ability to store and distribute Virtual
+Machine disks in the container image registry. Registry Disks can be assigned
+to Virtual Machines in the disks section of the Virtual Machine spec.
+
+## When to use a Registry Disk
+
+Registry Disks are ephemeral storage devices that can be assigned to any number
```
Comment (rmohr) [inline registry-disk.md]: I think it is clear, if one thinks about the container registry re-use, but could you add one or two sentences, which make it more explicit, that no shared storage is needed, after the images are pulled?

File: registry-disk.md (hunk 1 above)
Comment (davidvossel) [reply registry-disk.md]: nope, the only case where data will be reused is if the process was allowed to recover within the same virt-launcher cgroup.

File: registry-disk.md (hunk 2 above)
Comment (davidvossel) [reply registry-disk.md]: yup

File: registry-disk.md (hunk 3 above)
Comment (davidvossel) [reply registry-disk.md]: i'll add some more about this

**Pull request merged**